scipy/linalg/_basic.py

```diff
@@ -1081,7 +1081,7 @@ def lstsq(a, b, cond=None, overwrite_a=False, overwrite_b=False,
         Least-squares solution.
     residues : (K,) ndarray or float
         Square of the 2-norm for each column in ``b - a x``, if ``M > N`` and
-        ``ndim(A) == n`` (returns a scalar if ``b`` is 1-D). Otherwise a
+        ``rank(A) == n`` (returns a scalar if ``b`` is 1-D). Otherwise a
```

Comment:
So I think this should actually be "if M >= N and rank(A) == N".

Reply:
@ilayn do you agree on this suggestion?